internals/models.py

```diff
@@ -953,19 +953,79 @@ def get_in_milestone(
     q = Feature.query()
     q = q.order(Feature.name)
     q = q.filter(Feature.shipped_milestone == milestone)
+    q = q.filter(Feature.impl_status_chrome.IN([ENABLED_BY_DEFAULT, DEPRECATED, REMOVED]))
     desktop_shipping_features = q.fetch(None)
 
     # Features with an android shipping milestone but no desktop milestone.
     q = Feature.query()
     q = q.order(Feature.name)
     q = q.filter(Feature.shipped_android_milestone == milestone)
     q = q.filter(Feature.shipped_milestone == None)
+    q = q.filter(Feature.impl_status_chrome.IN([ENABLED_BY_DEFAULT, DEPRECATED, REMOVED]))
     android_only_shipping_features = q.fetch(None)
 
+    # Features that are in origin trial (Desktop) in this milestone
+    q = Feature.query()
+    q = q.filter(ndb.AND(Feature.ot_milestone_desktop_start != None, Feature.ot_milestone_desktop_start <= milestone))
+    q = q.filter(Feature.impl_status_chrome.IN([ORIGIN_TRIAL]))
+    desktop_origin_trial_features = q.fetch(None)
+    desktop_origin_trial_features = filter(lambda f: f.ot_milestone_desktop_end >= milestone, desktop_origin_trial_features)
+    desktop_origin_trial_features.sort(key=lambda f: f.name)
+    
+    # Features that are in origin trial (Android) in this milestone
+    q = Feature.query()
+    q = q.filter(ndb.AND(Feature.ot_milestone_android_start != None, Feature.ot_milestone_android_start <= milestone))
+    q = q.filter(Feature.ot_milestone_desktop_start == None)
+    q = q.filter(Feature.impl_status_chrome.IN([ORIGIN_TRIAL]))
+    android_origin_trial_features = q.fetch(None)
+    android_origin_trial_features = filter(lambda f: f.ot_milestone_android_end >= milestone, android_origin_trial_features)
+    android_origin_trial_features.sort(key=lambda f: f.name)
+
+    # Features that are in dev trial (Desktop) in this milestone
+    q = Feature.query()
+    q = q.filter(ndb.AND(Feature.dt_milestone_desktop_start != None, Feature.dt_milestone_desktop_start == milestone))
+    q = q.filter(Feature.impl_status_chrome.IN([BEHIND_A_FLAG]))
+    desktop_dev_trial_features = q.fetch(None)
+
+    q = Feature.query()
+    q = q.order(Feature.name)
+    q = q.filter(Feature.shipped_milestone == milestone) # Old Entries that do not have dt_milestone_start field 
```

Comment:
I like the idea of adding in legacy features that do not have a DT milestone.  However, I don't think that this is the right logic.   For example, if the user scrolls back in time to previous milestones, the same BEHIND_A_FLAG features would appear again and again in every milestone.   I think the missing condition is that these legacy features should only be considered if the milestone in this API request is the one on the stable channel currently.  There is no need to check shipped_milestone.